dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task201008LoadAppsSecrets.java

```diff
@@ -0,0 +1,143 @@
+        final Key key = generateKey(loadPass(null));
+        try {
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(serverDir,
+                    entry -> matchesFileName(entry.getFileName().toString()))) {
+                for (final Path path : stream) {
+                    try {
+                        final Map<String, List<AppSecrets>> secretsBySiteId = importSecrets(
+                                path, key);
+                        for (final Entry<String, List<AppSecrets>> importEntry : secretsBySiteId
+                                .entrySet()) {
+                            final String siteId = importEntry.getKey();
+                            if (Host.SYSTEM_HOST.equalsIgnoreCase(siteId) || findSite(siteId,
+                                    dotConnect)) {
+                                final List<AppSecrets> appSecrets = importEntry.getValue();
+                                for (final AppSecrets secrets : appSecrets) {
+                                    if (!secrets.getSecrets().isEmpty()) {
+                                        final AppDescriptor descriptor = descriptorsMap
+                                                .get(secrets.getKey().toLowerCase());
+                                        if (null != descriptor) {
+                                            validateForSave(mapForValidation(secrets), descriptor);
+                                            final char[] chars = toJsonAsChars(secrets);
+                                            final String internalKey = internalKey(secrets.getKey(),
+                                                    siteId);
+                                            keyStoreHelper.saveValue(internalKey, chars);
+                                            importCount++;
+                                        } else {
+                                            throw new DotDataException(String.format(
+                                                    "I cant't import a secret meant for a descriptor `%s` that doesn't exist locally.",
+                                                    secrets.getKey()));
+                                        }
+                                    } else {
+                                        throw new DotDataException(
+                                                "I cant't import a secret that was generated empty.");
+                                    }
+                                }
+                            } else {
+                                throw new DotDataException(String.format(
+                                        "I cant't import a secret that belongs into a unknown site `%s`.",
+                                        siteId));
+                            }
+                        }
+                    } finally {
+                        path.toFile().delete();
+                    }
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)